packages/graph/hash_graph/lib/graph/src/knowledge/entity/query.rs

```diff
@@ -63,7 +66,155 @@ impl RecordPath for EntityQueryPath<'_> {
 impl<'q> TryFrom<Path> for EntityQueryPath<'q> {
     type Error = Report<de::value::Error>;
 
-    fn try_from(_path: Path) -> Result<Self, Self::Error> {
-        todo!("https://app.asana.com/0/1203007126736607/1203167266370358/f")
+    fn try_from(path: Path) -> Result<Self, Self::Error> {
+        Self::deserialize(de::value::SeqDeserializer::new(
+            path.segments.into_iter().map(|segment| segment.identifier),
+        ))
+        .into_report()
+    }
+}
+
+/// A single token in a [`EntityQueryPath`].
+#[derive(Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub enum EntityQueryToken {
+    Id,
+    OwnedById,
+    CreatedById,
+    UpdatedById,
+    RemovedById,
+    Version,
+    Type,
+    Properties,
+    IncomingLinks,
+    OutgoingLinks,
+}
+
+/// Deserializes a [`EntityQueryPath`] from a string sequence.
+pub struct EntityQueryPathVisitor {
+    /// The current position in the sequence when deserializing.
+    position: usize,
+}
+
+impl EntityQueryPathVisitor {
+    #[must_use]
+    pub const fn new(position: usize) -> Self {
+        Self { position }
+    }
+}
+
+impl<'de> Visitor<'de> for EntityQueryPathVisitor {
+    type Value = EntityQueryPath<'de>;
+
+    fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        formatter.write_str(
+            "a sequence consisting of `ownedById`, `version`, `type`, `properties`, \
+             `incomingLinks, or `outgoingLinks`",
+        )
+    }
+
+    fn visit_seq<A>(mut self, mut seq: A) -> Result<Self::Value, A::Error>
+    where
+        A: SeqAccess<'de>,
+    {
+        let token = seq
+            .next_element()?
+            .ok_or_else(|| de::Error::invalid_length(self.position, &self))?;
+        self.position += 1;
+        Ok(match token {
+            EntityQueryToken::Id => EntityQueryPath::Id,
+            EntityQueryToken::OwnedById => EntityQueryPath::OwnedById,
+            EntityQueryToken::CreatedById => EntityQueryPath::CreatedById,
+            EntityQueryToken::UpdatedById => EntityQueryPath::UpdatedById,
+            EntityQueryToken::RemovedById => EntityQueryPath::RemovedById,
+            EntityQueryToken::Version => EntityQueryPath::Version,
+            EntityQueryToken::Type => {
+                let entity_type_query_path =
+                    EntityTypeQueryPathVisitor::new(self.position).visit_seq(seq)?;
+
+                EntityQueryPath::Type(entity_type_query_path)
+            }
+            EntityQueryToken::Properties => {
+                let property = seq
+                    .next_element()?
+                    .ok_or_else(|| de::Error::invalid_length(self.position, &self))?;
+                EntityQueryPath::Properties(Some(property))
+            }
+            EntityQueryToken::OutgoingLinks | EntityQueryToken::IncomingLinks => {
+                todo!("https://app.asana.com/0/0/1203167266370359/f")
+            }
+        })
+    }
+}
+impl<'de: 'k, 'k> Deserialize<'de> for EntityQueryPath<'k> {
```

Comment:
What is `'k` here? Just wondering because we use `'q` on the type

Reply:
I used `'k` before and forgot to update this here.